### ISSUE TITLE
Allow permutation starting with 0 in Thurston-Veech construction

### ISSUE THREAD

**Issue by videlec** [flatsurf/sage-flatsurf#133]
It would be convenient if the constructor of `ThurstonVeech` would accept permutations starting at zero. There could be, as with `surface-dynamics` origami the `as_tuple` keyword.
```
sage: from surface_dynamics import Origami
sage: Origami([2,1,3], [1,3,2])
(1,2)(3)
(1)(2,3)
sage: Origami([1,0,2], [0,2,1], as_tuple=True)
(1,2)(3)
(1)(2,3)
```

**Comment by saraedum** [comment]
@videlec I find `as_tuple` actually somewhat misleading in surface-dynamics. The doctest there has:

        sage: Origami('(0)(1,3)(2)', '(0,3,2,1)', as_tuple=True)
        (1)(2,4)(3)
        (1,4,3,2)

So here `as_tuple` just means zero-based (from the user's perspective.)

Also, there is this oddity in parsing:

```
sage: Origami((0,1,2,3), [0,1,2,3], as_tuple=True)
Warning: the origami is not connected
(1)(2)(3)(4)
(1)(2)(3)(4)
sage: Origami((1,2,3,4), [1,2,3,4], as_tuple=False)
(1,2,3,4)
(1)(2)(3)(4)
```

For `PermutationGroupElement`, there's a difference between a tuple (a cycle) and a list (the image.)

This is all a bit confusing and maybe we don't want to replicate this or improve the situation at surface-dynamics while we're at it?

**Comment by videlec** [comment]
I agree that this is confusing but I did not find a convincing interface. Internally everything is zero based and it is much better this way because Python is zero based. However, sage persists in using permutations starting with 1 which drove me to use sometimes the 1 based convention.

I am open to

1. a better design to mix 0 based and 1 based
2. use 0 based everywhere (except maybe for sagemath `Permutation` and `PermutationGroupElement` input)
3. other propositions
